src/ape/api/providers.py

```diff
@@ -184,6 +184,38 @@ def ws_uri(self) -> Optional[str]:
         """
         return None
 
+    @property
+    def settings(self) -> PluginConfig:
+        """
+        The combination of settings from ``ape-config.yaml`` and ``.provider_settings``.
+        """
+        CustomConfig = self.config.__class__
+        data = {**self.config.dict(), **self.provider_settings}
+        return CustomConfig.parse_obj(data)
+
+    @property
+    def connection_id(self) -> Optional[str]:
+        """
+        A connection ID for this provider's connection used for managing
+        multiple connections to providers and identifying them, especially if
+        using multiple providers of the same type, such as multiple Geth --dev
+        nodes.
+        """
+
+        try:
+            chain_id = self.chain_id
+        except Exception:
+            if chain_id := self.settings.get("chain_id"):
+                pass
+
+            else:
+                # A connection is required to obtain a chain ID for this provider.
+                return None
+
+        # NOTE: If other provider settings are different, ``.update_settings()``
+        #    should be called.
+        return f"{self.network_choice}:-{chain_id}"
```

Comment:
I think chain ID won't change if you have two different fork nodes of the same network, only the port changes

Reply:
you can make it change tho

Reply:
but ill make the SubprocessProvider handle this , that is a good call out.

Reply:
Now, the command affects the ID, so if you run multiple nodes with different commands, it recognized it is different.

If a provider doesn't include the port in the command, itll have to handle independently in it's plugin implementation code.